### PR TITLE
Modinvosto 2 define ap is

### PR DESCRIPTION
**Pull request by pautau-s**
## Purpose
[MODINVOSTO-2](https://issues.folio.org/browse/MODINVOSTO-2) Define the basic invoice storage APIs and generate stub code via RMB

## Approach
- RAML file is defined
- Module and Deployment descriptors defined
- added acq-models sub-module with schemas defined in [folio-org/acq-models#99](https://github.com/folio-org/acq-models/pull/99)
- Pom incorporates RMB code generation from RAML/schemas.
- DB schema script defined to create 2 tables

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -2,9 +2,71 @@
...
+          "methods": ["DELETE"],
+          "pathPattern": "/invoice-storage/invoice-lines/{id}",
+          "permissionsRequired": ["invoice-lines.item.delete"]
+        }
+      ]
+    },
     {
       "id": "_tenant",
-      "version": "1.2",
+      "version": "1.0",
```
Comment (craigmcnally) [inline descriptors/ModuleDescriptor-template.json]: please use 1.2.  If we ever want to be able to load reference/sample data via the new preferred mechanism (tenant API), we'll need at 1.2

File: pom.xml [hunk 2]
```diff
@@ -12,449 +12,351 @@
...
+        <executions>
+          <execution>
+            <goals>
+              <goal>compile</goal>
+            </goals>
+          </execution>
+        </executions>
         <dependencies>
-            <dependency>
-                <groupId>io.vertx</groupId>
```
Comment (craigmcnally) [inline pom.xml]: Pretty sure we're going to need this, but we can add it later

File: pom.xml [hunk 3]
```diff
@@ -12,449 +12,351 @@
...
-                <artifactId>vertx-stack-depchain</artifactId>
-                <version>${vertx.stack.depchain.version}</version>
-                <type>pom</type>
-                <scope>import</scope>
-            </dependency>
-            <dependency>
-                <groupId>com.fasterxml.jackson</groupId>
```
Comment (craigmcnally) [inline pom.xml]: same as above... will probably need this eventually

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -2,6 +2,68 @@
...
   "name": "Invoice CRUD module",
   "provides": [
+    {
+      "id": "invoice-storage.invoice",
+      "version": "1.0",
+      "handlers": [
+        {
+          "methods": ["GET"],
+          "pathPattern": "/invoice-storage/invoices",
+          "permissionsRequired": ["invoice.collection.get"]
```
Comment (piotr-kalashuk) [inline descriptors/ModuleDescriptor-template.json]: Please use `invoice-storage.` prefix (as you already defined in `permissionSets`)

File: pom.xml [hunk 5]
```diff
@@ -12,449 +12,370 @@
...
-        <maven.resources.plugin.version>3.0.1</maven.resources.plugin.version>
-        <copy.rename.maven.plugin.version>1.0</copy.rename.maven.plugin.version>
-        <maven.antrun.plugin.version>1.8</maven.antrun.plugin.version>
-        <properties.maven.plugin.version>1.0.0</properties.maven.plugin.version>
-        <maven.shade.plugin.version>2.3</maven.shade.plugin.version>
-        <maven.release.plugin.version>2.5.3</maven.release.plugin.version>
-        <maven.surefire.plugin>2.22.1</maven.surefire.plugin>
-    </properties>
+  <properties>
+    <raml-module-builder.version>23.3.0</raml-module-builder.version>
```
Comment (piotr-kalashuk) [inline pom.xml]: Please use the most recent [RMB release](https://github.com/folio-org/raml-module-builder/releases) version (a the moment it is 23.9.0)

File: pom.xml [hunk 6]
```diff
@@ -12,449 +12,370 @@
...
-            <name>FOLIO Maven Repository</name>
-            <url>https://repository.folio.org/repository/maven-folio</url>
-        </repository>
-    </repositories>
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>io.vertx</groupId>
+        <artifactId>vertx-stack-depchain</artifactId>
+        <version>3.5.4</version>
```
Comment (piotr-kalashuk) [inline pom.xml]: Please use `vertx.version` property

File: ramls/invoice.raml [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+          description: The UUID of an invoice line
+          type: UUID
+      type:
+        collection-item:
+          schema: invoiceLine
+          exampleItem: !include acq-models/mod-invoice-storage/examples/invoice_line.sample
+      put:
+        is: [validate]
+
+  /invoice_number:
```
Comment (piotr-kalashuk) [inline ramls/invoice.raml]: Please use `-` instead of `_`

File: src/main/java/org/folio/rest/impl/InvoiceStorageImpl.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+import static io.vertx.core.Future.succeededFuture;
+
+public class InvoiceStorageImpl implements InvoiceStorage {
+
+  public static final String NOT_SUPPORTED = "Not supported";
+
+  @Validate
+  @Override
+  public void getInvoiceStorageInvoices(int offset, int limit, String query, String lang, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    asyncResultHandler.handle(succeededFuture(GetInvoiceStorageInvoicesResponse.respond500WithTextPlain(NOT_SUPPORTED)));
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/InvoiceStorageImpl.java]: You can use `Response.status(501).build()` i.e. `501 (Not Implemented)` for now

File: src/test/java/org/folio/rest/imp/InvoicesTest.java [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+import java.util.Locale;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static io.restassured.RestAssured.given;
+import static org.folio.rest.RestVerticle.OKAPI_HEADER_TENANT;
+
+public class InvoicesTest {
```
Comment (piotr-kalashuk) [inline src/test/java/org/folio/rest/imp/InvoicesTest.java]: Probably it makes sense to use test suites here from the beginning to separate tests logically to different classes

Comment (pautau-s) [reply src/test/java/org/folio/rest/imp/InvoicesTest.java]: Done

File: src/main/java/org/folio/rest/impl/InvoiceStorageImpl.java (hunk 8 above)
Comment (pautau-s) [reply src/main/java/org/folio/rest/impl/InvoiceStorageImpl.java]: Done

File: ramls/invoice.raml (hunk 7 above)
Comment (pautau-s) [reply ramls/invoice.raml]: Corrected

File: pom.xml (hunk 6 above)
Comment (pautau-s) [reply pom.xml]: Done

File: descriptors/ModuleDescriptor-template.json (hunk 4 above)
Comment (pautau-s) [reply descriptors/ModuleDescriptor-template.json]: Corrected

File: pom.xml (hunk 5 above)
Comment (pautau-s) [reply pom.xml]: Done

File: src/main/java/org/folio/rest/impl/InvoiceStorageImpl.java [hunk 10]
```diff
@@ -20,71 +20,71 @@
   @Validate
   @Override
   public void getInvoiceStorageInvoices(int offset, int limit, String query, String lang, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    asyncResultHandler.handle(succeededFuture(GetInvoiceStorageInvoicesResponse.respond500WithTextPlain(NOT_SUPPORTED)));
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/InvoiceStorageImpl.java]: `NOT_SUPPORTED` constant can be removed

**Pull request merged**